xarray/core/dataset.py

```diff
@@ -2637,8 +2642,9 @@ def chunk(
             )
             chunks = {}
 
-        if isinstance(chunks, (Number, str, int)):
-            chunks = dict.fromkeys(self.dims, chunks)
+        if not isinstance(chunks, Mapping):
+            # We need to ignore since mypy doesn't recognize this can't be `None`
```

Comment:
Wait... is the above check for None correct? I'm not sure that the chunks_kwargs can be None, isn't it always a dict? So this would make the warning not reachable...

Reply:
I don't think `chunks_kwargs` can be `None`, but `chunks` can... Does that answer your Q?

Reply:
Well, it answers the question.
But then this means that the above code is wrong because the if statement is never reachable...
Doesn't need to be fixed in this PR, but maybe rewriting this will remove the need for the type ignore.

Reply:
Oh I see what you mean...